src/chat_manager.ts

```diff
@@ -1,32 +1,32 @@
-import {
-  BaseClient,
-  HOST_BASE,
-  Instance,
-  Logger,
-  TokenProvider,
-} from 'pusher-platform';
+import { BaseClient, HOST_BASE, Instance, Logger } from 'pusher-platform';
 
+import BasicCursor from './basic_cursor';
 import ChatManagerDelegate from './chat_manager_delegate';
 import CurrentUser from './current_user';
 import GlobalUserStore from './global_user_store';
+import PayloadDeserializer from './payload_deserializer';
+import TokenProvider from './token_provider';
```

Comment:
I'm not sure changing this is what we want. 

If we want to change it because it matches the new TokenProvider interface we want, then we should make and provide an interface and then provide what our current `TokenProvider` (in chatkit-client-js) does as a default implementation of that interface.

Reply:
Gone back to using the platform `TokenProvider`, and then checking with `instanceof` whether to set the `userId`, I think this is better?

Reply:
Looks good to me. Just realised - should this line https://github.com/pusher/chatkit-client-js/pull/16/files#diff-1764d4e0007db0e880e7426d753dd666L24 have an `implements TokenProvider` in it, where `TokenProvider` there would be the platform SDK's `TokenProvider`? Can't remember how explicit you should be with interface implementations in TypeScript

Reply:
Ahh yeah, I think it probably _should_ but it isn't a big deal that it doesn't. My understanding is that the `implements` only gives you compile time checks that you've implemented the interface properly, it doesn't actually make any difference to the type itself (since typescript stuff is all structurally typed). Still, we should be explicit :+1: